### PR TITLE
fix: Pass app_profile_id when building updated request

### PR DESCRIPTION
**Pull request by Mariatta**
- Pass the app_profile_id from the message
- Update unittest

Fixes internal bug #214449800

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: google/cloud/bigtable/row_data.py [hunk 1]
```diff
@@ -627,6 +628,7 @@ def build_updated_request(self):
         r_kwargs = {
             "table_name": self.message.table_name,
             "filter": self.message.filter,
+            "app_profile_id": self.message.app_profile_id,
```
Comment (igorbernstein2) [inline google/cloud/bigtable/row_data.py:630]: I think this works, but it puts us in the same position as before: if ReadRowsRequest ever adds a new field, we will end up dropping that field on retry. I think one of 2 things should happen:
1. clone the entry request and overwrite the RowSet & limit. This way if new fields are added, we are sure to copy them
2. Add a unit test that ensure no new fields are added with a comment to instructing someone to update this code

I think 1 is definitely preferable

**Review by igorbernstein2 — CHANGES_REQUESTED** [review]

**Review by kolea2 — APPROVED** [review]

File: tests/unit/test_row_data.py [hunk 2]
```diff
@@ -902,7 +905,10 @@ def test_RRRM_build_updated_request(rrrm_data):
     result = request_manager.build_updated_request()
 
     expected_result = _ReadRowsRequestPB(
-        table_name=TABLE_NAME, filter=row_filter.to_pb(), rows_limit=6
+        table_name=TABLE_NAME,
```
Comment (kolea2) [inline tests/unit/test_row_data.py:908]: is it easy to also check this in the system tests?

Comment (Mariatta) [reply tests/unit/test_row_data.py:908]: I just ran the `system` test suite, but the ``build_updated_request`` doesn't seem to be covered there currently.

**Review by igorbernstein2 — APPROVED** [review]
lgtm

**Pull request merged**